File: Client/orgTable/code/ProtocolWarpStone.cs
using System;
using System.Text;

namespace Protocol
{
	public class WarpStoneInfo : Protocol.IProtocolStream
	{
		public UInt32 id;
		public byte level;
		public UInt32 energy;

		#region METHOD

			public void encode(byte[] buffer, ref int pos_)
			{
				BaseDLL.encode_uint32(buffer, ref pos_, id);
				BaseDLL.encode_int8(buffer, ref pos_, level);
				BaseDLL.encode_uint32(buffer, ref pos_, energy);
			}

			public void decode(byte[] buffer, ref int pos_)
			{
				BaseDLL.decode_uint32(buffer, ref pos_, ref id);
				BaseDLL.decode_int8(buffer, ref pos_, ref level);
				BaseDLL.decode_uint32(buffer, ref pos_, ref energy);
			}
		#endregion

	}

	public class ChargeInfo : Protocol.IProtocolStream
	{
		public UInt32 id;
		public UInt32 num;

		#region METHOD

			public void encode(byte[] buffer, ref int pos_)
			{
				BaseDLL.encode_uint32(buffer, ref pos_, id);
				BaseDLL.encode_uint32(buffer, ref pos_, num);
			}

			public void decode(byte[] buffer, ref int pos_)
			{
				BaseDLL.decode_uint32(buffer, ref pos_, ref id);
				BaseDLL.decode_uint32(buffer, ref pos_, ref num);
			}
		#endregion

	}

	[Protocol]
	public class SceneWarpStoneUnlockReq : Protocol.IProtocolStream, Protocol.IGetMsgID
	{
		public const UInt32 MsgID = 506901;
		public UInt32 id;

		#region METHOD
			public UInt32 GetMsgID()
			{
				return MsgID;
			}

			public void encode(byte[] buffer, ref int pos_)
			{
				BaseDLL.encode_uint32(buffer, ref pos_, id);
			}

			public void decode(byte[] buffer, ref int pos_)
			{
				BaseDLL.decode_uint32(buffer, ref pos_, ref id);
			}
		#endregion

	}

	[Protocol]
	public class SceneWarpStoneUnlockRes : Protocol.IProtocolStream, Protocol.IGetMsgID
	{
		public const UInt32 MsgID = 506902;
		public UInt32 result;

		#region METHOD
			public UInt32 GetMsgID()
			{
				return MsgID;
			}

			public void encode(byte[] buffer, ref int pos_)
			{
				BaseDLL.encode_uint32(buffer, ref pos_, result);
			}

			public void decode(byte[] buffer, ref int pos_)
			{
				BaseDLL.decode_uint32(buffer, ref pos_, ref result);
			}
		#endregion

	}

	[Protocol]
	public class SceneWarpStoneChargeReq : Protocol.IProtocolStream, Protocol.IGetMsgID
	{
		public const UInt32 MsgID = 506903;
		public UInt32 id;
		public ChargeInfo[] info = new ChargeInfo[0];

		#region METHOD
			public UInt32 GetMsgID()
			{
				return MsgID;
			}

			public void encode(byte[] buffer, ref int pos_)
			{
				BaseDLL.encode_uint32(buffer, ref pos_, id);
				BaseDLL.encode_uint16(buffer, ref pos_, (UInt16)info.Length);
				for(int i = 0; i < info.Length; i++)
				{
					info[i].encode(buffer, ref pos_);
				}
			}

			public void decode(byte[] buffer, ref int pos_)
			{
				BaseDLL.decode_uint32(buffer, ref pos_, ref id);
				UInt16 infoCnt = 0;
				BaseDLL.decode_uint16(buffer, ref pos_, ref infoCnt);
				info = new ChargeInfo[infoCnt];
				for(int i = 0; i < info.Length; i++)
				{
					info[i] = new ChargeInfo();
					info[i].decode(buffer, ref pos_);
				}
			}
		#endregion

	}

	[Protocol]
	public class SceneWarpStoneChargeRes : Protocol.IProtocolStream, Protocol.IGetMsgID
	{
		public const UInt32 MsgID = 506904;
		public UInt32 result;

		#region METHOD
			public UInt32 GetMsgID()
			{
				return MsgID;
			}

			public void encode(byte[] buffer, ref int pos_)
			{
				BaseDLL.encode_uint32(buffer, ref pos_, result);
			}

			public void decode(byte[] buffer, ref int pos_)
			{
				BaseDLL.decode_uint32(buffer, ref pos_, ref result);
			}
		#endregion

	}

}

File: Client/orgTable/code/ProtocolWarpStone.lua
local descriptor = require "descriptor"

module "ProtocolWarpStone"

WarpStoneInfo = descriptor.def_struct("WarpStoneInfo", 
	{
		descriptor.def_scalar_field("id", descriptor.type_uint32, 0),
		descriptor.def_scalar_field("level", descriptor.type_uint8, 0),
		descriptor.def_scalar_field("energy", descriptor.type_uint32, 0),
	}
)

ChargeInfo = descriptor.def_struct("ChargeInfo", 
	{
		descriptor.def_scalar_field("id", descriptor.type_uint32, 0),
		descriptor.def_scalar_field("num", descriptor.type_uint32, 0),
	}
)

SceneWarpStoneUnlockReq = descriptor.def_message("SceneWarpStoneUnlockReq", 506901, 
	{
		descriptor.def_scalar_field("id", descriptor.type_uint32, 0),
	}
)

SceneWarpStoneUnlockRes = descriptor.def_message("SceneWarpStoneUnlockRes", 506902, 
	{
		descriptor.def_scalar_field("result", descriptor.type_uint32, 0),
	}
)

SceneWarpStoneChargeReq = descriptor.def_message("SceneWarpStoneChargeReq", 506903, 
	{
		descriptor.def_scalar_field("id", descriptor.type_uint32, 0),
		descriptor.def_message_vector_field("info", ChargeInfo),
	}
)

SceneWarpStoneChargeRes = descriptor.def_message("SceneWarpStoneChargeRes", 506904, 
	{
		descriptor.def_scalar_field("result", descriptor.type_uint32, 0),
	}
)

